libtcmu.c

```diff
@@ -913,6 +912,7 @@ static void remove_device(struct tcmulib_context *ctx,
 	if (ret < 0) {
 		tcmu_err("could not cleanup mailbox lock %s: %d\n", dev_name, errno);
 	}
+	dev->handler->removed(dev);
```

Comment:
https://github.com/open-iscsi/tcmu-runner/pull/69

This patch above moved dev->handler->removed(dev); to end to fix freeing invalid fd.